documentation/smg/04_building_with_cmake.md

```diff
@@ -1,38 +1,151 @@
 # Building with CMake
 
+## Build dependencies
+
+A developer building Herbert/Horace will need to install:
+
+- [Matlab](https://www.mathworks.com/products/matlab.html) >= 2018b
+  - on Linux the desired Matlab version must be on the path when running the
+  script
+  - on Windows, by default, the latest version is found via the registry
+- [CMake](https://cmake.org/download/) >= 3.7
+  - must be on the path for both Windows and Linux
+- System compiler:
+  - Linux: [GCC](https://gcc.gnu.org/) >= 6.3.0
+  - Windows:
+  [Visual Studio build tools](https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019)
+  \>= 2017
+
+## Using build scripts
+
+Horace and Herbert contain PowerShell and Bash scripts named `build.[ps1|sh]`.
+These scripts provide an easy way to build Herbert/Horace on Windows or Linux.
+Theses same scripts are used to build Horace and Herbert on the CI servers,
+so they are tested regularly.
+
+A developer can build using default options with the `--build` flag.
+
+From Herbert/Horace's root directory:
+
+PowerShell:
+
+```powershell
+./tools/build_config/build.ps1 -build
+```
+
+Bash:
+
+```bash
+./tools/build_config/build.sh --build
+```
+
+_(Note that Bash uses two dashes (`--`) whilst PowerShell uses one.)_
+
+Generally, the default build options will be sufficient,
+but there are several configurable parameters.
+Both scripts have a `help` flag which display the available actions:
+
+PowerShell:
+
+```powershell
+./tools/build_config/build.ps1 -help
+```
+
+Bash:
+
+```bash
+./tools/build_config/build.sh --help
+```
+
+## Using CMake directly
+
 CMake provides a way to generate build files across multiple platforms. It also
 ships with a script to aid in compiling Matlab mex libraries.
 
+The following process is roughly what the build scripts automate.
```

Comment:
Again, checking the reader knows about the build scripts. If it is what you just described above, a confirmatory `(see above)` could reassure the reader what you are referring to.